### DISCUSSION TITLE
How to use elementInSupportOf function to get the correct result of an element？

### DISCUSSION THREAD

**Discussion by Rennngw** [gismo/gismo#587] [Q&A] [answered]
My example code is as follows:

```
	auto bSpline = gsNurbsCreator<double>::BSplineCube(20, 0, 0, 0);
	bSpline->degreeElevate();
    for (int i = 0; i < 2; ++i)
		bSpline->uniformRefine();

	gsInfo << "bsplinebasis" << bSpline->basis();

       for (auto domIt = bSpline->basis().makeDomainIterator(); domIt->good(); domIt->next())
	{
		gsVector<double> center = domIt->centerPoint();
		size_t index = bSpline->basis().elementIndex(center);
		gsInfo << bSpline->basis().elementInSupportOf(index);
	}
```

I think the result obtained is not correct, how can I get the correct interval result.

**The result i got is:**
![image](https://user-images.githubusercontent.com/110526351/218251786-45f9532f-afc0-40c7-9d74-5fb0d8c37bb2.png)
which means:
first element: [0 0.25]x[0 0.25]x[0 0.25]
second element:[0.25 0.5]x[0 0.25]x[0 0.25]
third element:[0.25 0.5]x[0 0.25]x[0 0.25]
...
**third element is same as second**.is it wrong?

**I think the correct result is:**
first element: [0 0.25]x[0 0.25]x[0 0.25]
second element:[0.25 0.5]x[0 0.25]x[0 0.25]
third element:[0.5 0.75]x[0 0.25]x[0 0.25]
...

I'm not sure whether i'm right or wrong.
Thank you for your help in advance.

Best regards
Rennngw

**Comment by ckarampa** [comment, marked as answer]
Hello, first of all I would like to suggest that you add "/n/n" at the end of your prints , especially when posting questions to make it easier for someone to understand.


So what you're doing is working fine, but I don't think it is what you were looking for. As stated in the documentation `elementInSupportOf` returns an element ( not a specific one ) in the support of the basis you are indexing. I am guessing that you wanted to get the corners of every element. This can be done as follows:
```c++
    auto bSpline = gsNurbsCreator<double>::BSplineCube(20, 0, 0, 0);
	bSpline->degreeElevate();
    for (int i = 0; i < 2; ++i)
		bSpline->uniformRefine();

	gsInfo << bSpline->basis() << "\n\n";

    for (auto domIt = bSpline->basis().makeDomainIterator(); domIt->good(); domIt->next())
	{
		gsInfo << domIt->lowerCorner().transpose() << "\n" << domIt->upperCorner().transpose()  << "\n\n";  
	}

    return 0; 
```

**Reply by ckarampa** [reply]
@Rennngw can you please confirm if your question has been answered, so that I can close the discussion?

**Reply by Rennngw** [reply]
Yes, please.
